addons/web/static/tests/views/list_view_tests.js

```diff
@@ -259,6 +259,139 @@ QUnit.module("Views", (hooks) => {
         assert.isNotVisible(target.querySelector(".o_list_button_discard"));
     });
 
+    QUnit.test("select record range with shift click", async function (assert) {
+        await makeView({
+            type: "list",
+            resModel: "foo",
+            serverData,
+            arch: '<tree><field name="foo"/><field name="int_field"/></tree>',
+        });
+        await click(target.querySelectorAll(".o_data_row .o_list_record_selector input")[0]);
```

Comment:
even if it seems more complicated here, same nitpicking comment applies: you could probably do something like
```suggestion
        await click(target, ".o_data_row[data-id=datapoint_42] .o_list_record_selector input");
```